include/circt/Dialect/FIRRTL/FIRRTLTypesImpl.td

```diff
@@ -472,15 +472,15 @@ def BaseTypeAliasImpl : FIRRTLImplType<"BaseTypeAlias", [DeclareTypeInterfaceMet
       // FIRRTLBaseType utils.
       FIRRTLBaseType getAnonymousType();
       FIRRTLBaseType getPassiveType();
-      FIRRTLBaseType getConstType(bool isConst);
+      FIRRTLBaseType getConstType(bool isConst) const;
       FIRRTLBaseType getAllConstDroppedType();
 
       /// Return the recursive properties of the type.
       RecursiveTypeProperties getRecursiveTypeProperties() const;
 
       // If a given `newInnerType` is identical to innerType, return `*this`
       // because we can reuse the type alias. Otherwise return `newInnerType`.
-      FIRRTLBaseType getModifiedType(FIRRTLBaseType newInnerType);
+      FIRRTLBaseType getModifiedType(FIRRTLBaseType newInnerType) const;
```

Comment:
Maybe const-qualify these separately?

What's this for, just curious?

https://mlir.llvm.org/docs/Rationale/UsageOfConst/

Anyway separate commit works for me (as already done), thanks!

Reply:
Yeah, I needed these when zipping together the enum types, which gives constant references to the elements, which I then need to be able to convert to the const version of the type.  I might have been able to just add a non-const iterator to the enum type, but this seemed a bit better to me.